backend/server.py

```
@@ -0,0 +1,101 @@
+# App endpoints
+# ---------------------------------
+
+@app.get("/chat")
+async def chat(query: str, chat_id: str):
+    sources = [] # TODO: get authenticated sources from QuestDB
+
+    vectorstore = app.state.vectorstore
+    reranker = app.state.reranker
+    questdb_pool = app.state.questdb_pool
+
+    messages, _, _, lang_code = prepare_rag_context(query, questdb_pool, vectorstore, reranker, sources)
```

Ruff (0.14.14)

[error] `prepare_rag_context` may be undefined, or defined from star imports

(F405)